### PR TITLE
Refactor: force json for requests

### PR DESCRIPTION
**Pull request by rbienvenu0**
Why: forces to decode request content as json even if the content type isn't defined

WAZO-4352

---

> [!NOTE]
> Introduces stricter JSON handling and validation for HTTP endpoints.
> 
> - Force JSON parsing in `ConfigResource.patch` via `request.get_json(force=True)`
> - Add integration tests ensuring empty bodies on `POST action/Command` and `PATCH config` return `400`, with shared helper to issue raw HTTP requests
> - Update `CHANGELOG.md` to document the new parsing behavior and error responses
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 5c4396e4645fdc5cd6c6c66822c07fee5c600029. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: integration_tests/suite/test_config.py [hunk 1]
```diff
@@ -45,3 +46,27 @@ def test_update_config(self) -> None:
...
+        )
+        assert response.status_code == 400
+
+        response = requests.patch(
+            url,
+            headers=headers,
+            data=None,
+            verify=False,
+        )
+        assert response.status_code == 400
```
Comment (DrPyser) [inline integration_tests/suite/test_config.py]: Think it's worth it to simplify this pattern with some abstraction to make this more concise.

```
# this could be in a base class to reuse in all tests
def _make_raw_authenticated_patch(self, url, body):
    headers = {
        'X-Auth-Token': VALID_TOKEN,
    }

    return requests.patch(url, data=body, headers=headers, verify=False)

def test_that_empty_body_returns_400(self):
    port = ...
    url = ...
    values = ['', None]
    for value in values:
        response = self._make_raw_authenticated_patch(url, value)
        assert response.status_code == 400

```

File: integration_tests/suite/test_config.py [hunk 2]
```diff
@@ -45,3 +46,60 @@ def test_update_config(self) -> None:
...
+
+        response = requests.post(
+            action_url,
+            headers=headers,
+            data=None,
+            verify=False,
+        )
+        assert response.status_code == 400
```
[the remark this answers is not in the document]
Comment (rbienvenu0) [reply integration_tests/suite/test_config.py]: This particular test is gone

File: integration_tests/suite/test_config.py (hunk 1 above)
Comment (rbienvenu0) [reply integration_tests/suite/test_config.py]: Updated

Comment (rbienvenu0) [reply integration_tests/suite/test_config.py]: Updated

File: integration_tests/suite/test_config.py [hunk 3]
```diff
@@ -45,3 +46,39 @@ def test_update_config(self) -> None:
...
+        )
+
+    def test_that_empty_body_returns_400(self) -> None:
+        port = self.asset_cls.service_port(9491, 'amid')
+        urls = [
+            ('patch', f'http://127.0.0.1:{port}/1.0/config'),
+            ('post', f'http://127.0.0.1:{port}/1.0/action/Command'),
+        ]
+
+        for url in urls:
```
Comment (DrPyser) [inline integration_tests/suite/test_config.py]: this test is in "test_config" module, which is for the /config endpoint. 
`/action/Command` API should not be tested here, but in its own test module.

Comment (DrPyser) [reply integration_tests/suite/test_config.py]: the generic code above can be shared through the `APIIntegrationTest` base class.

File: integration_tests/suite/test_http.py [hunk 4]
```diff
@@ -196,3 +196,17 @@ def test_given_no_ajam_when_http_request_then_status_code_503(self) -> None:
...
+            ('patch', f'http://127.0.0.1:{port}/1.0/config'),
+            ('post', f'http://127.0.0.1:{port}/1.0/action/Command'),
+        ]
+
+        for url in urls:
+            response = self._make_raw_http_call(url[0], url[1], '')
+            assert response.status_code == 400, f'Error with url: {url}'
+
+            response = self._make_raw_http_call(url[0], url[1], None)
+            assert response.status_code == 400, f'Error with url: {url}'
```
Comment (DrPyser) [inline integration_tests/suite/test_http.py]: this module tests the http AMI interface, so I would only test /action/Command here. /config has its own `test_config.py` module.

Comment (DrPyser) [reply integration_tests/suite/test_http.py]: this test should be named `test_given_empty_body_when_post_command_then_status_code_400`.

Comment (DrPyser) [reply integration_tests/suite/test_http.py]: also, would be nice to use the same generic utilities as in wazo-auth PR: https://github.com/wazo-platform/wazo-auth/pull/424/changes .
This test could be 1 line:
```
def test_given_empty_body_when_post_command_then_status_code_400(self) -> None:
    self.assert_empty_body_returns_400([("POST", "action/Command")])
```

Comment (rbienvenu0) [reply integration_tests/suite/test_http.py]: Updated

File: integration_tests/suite/test_http.py [hunk 5]
```diff
@@ -196,3 +196,8 @@ def test_given_no_ajam_when_http_request_then_status_code_503(self) -> None:
                     )
                 ),
             )
+
+        def test_that_empty_body_when_post_command_returns_400(
+            self: TestHTTPError,
+        ) -> None:
+            self.assert_empty_body_returns_400([('post', 'action/Command')])
```
[the remark this answers is not in the document]
Comment (rbienvenu0) [reply integration_tests/suite/test_http.py]: Fixed

File: integration_tests/suite/test_config.py (hunk 3 above)
Comment (rbienvenu0) [reply integration_tests/suite/test_config.py]: Updated

**Review by DrPyser — DISMISSED** [review]
LGTM, well done! Now this can serve as a template, same pattern can be applied everywhere.
